lib/mode.coffee

```diff
@@ -0,0 +1,234 @@
+XRegExp = require 'xregexp'
+sb_exec = require 'sb-exec'
+path = require 'path'
+
+pattern = XRegExp('(?<file>[^\n\r]+):(?<from_line>\\d+):(?<from_col>\\d+):\\s*\
+  (?<to_line>\\d+):(?<to_col>\\d+)\\s+\
+  ((?<error>error|fatal error)|(?<warning>warning)|(?<info>note|help)):\\s+\
+  (?<message>.+?)[\n\r]+($|(?=[^\n\r]+:\\d+))', 's')
+
+parseOldMessages = (output, disabledWarnings) ->
+  elements = []
+  XRegExp.forEach output, pattern, (match) ->
+    if match.from_col == match.to_col
+      match.to_col = parseInt(match.to_col) + 1
+    range = [
+      [match.from_line - 1, match.from_col - 1],
+      [match.to_line - 1, match.to_col - 1]
+    ]
+    level = if match.error then 'error'
+    else if match.warning then 'warning'
+    else if match.info then 'info'
+    else if match.trace then 'trace'
+    else if match.note then 'note'
+    element =
+      type: level
+      message: match.message
+      file: match.file
+      range: range
+    elements.push element
+  buildMessages elements, disabledWarnings
+
+parseJsonMessages = (messages, disabledWarnings) ->
+  elements = []
+  for input in messages
+    continue unless input and input.spans
```

Comment:
📝 [Linter messages](https://github.com/steelbrain/linter/wiki/Linter-API#messages) don't (currently) require a range, is this ignoring potentially useful messages just because they don't have a span?